packages/headless/src/controllers/core/tab/headless-core-tab.ts

```diff
@@ -126,6 +135,11 @@ export function buildCoreTab(engine: CoreEngine, props: TabProps): Tab {
     ...controller,
 
     select() {
+      if (props.options.clearFiltersOnTabChange) {
+        dispatch(deselectAllBreadcrumbs());
+        dispatch(deselectAllNonBreadcrumbs());
+        dispatch(updateQuery({q: ''}));
+      }
```

Comment:
Wondering if we should refactor and use `prepareForSearchWithQuery`
https://github.com/coveo/ui-kit/blob/2698979fa47de81882269389e505b023696ebe0c/packages/headless/src/features/search/search-actions.ts#L76-L77